pkg/package-server/provider/registry.go

```diff
@@ -242,6 +246,12 @@ func (p *RegistryProvider) List(namespace string) (*operators.PackageManifestLis
 	defer p.mu.RUnlock()
 
 	pkgs := []operators.PackageManifest{}
+	if namespace != metav1.NamespaceAll {
```

Comment:
Instead of querying the cluster, can't we just loop through `p.clients` and return an empty list if there's no client for that namespace?

Reply:
A request for a namespace that doesn't exist would still return `PackageManifests` from the "global" namespace.